Erroneous lines shouldn't be added

sid-code opened the pull request:
If I enter a line that doesn't compile, any subsequent lines fail due to this line sitting at the top of the temporary file. This change checks if the compiler died while compiling the line, and if it did, prevents it from being added to the buffer.

sid-code committed:
Erroneous lines shouldn't be added

sid-code committed:
Fix :delete command

sid-code committed:
Fix :delete command

sid-code committed:
Modify compilation instructions

rangeChecks are needed for :delete

sid-code committed:
Modify compilation instructions

rangeChecks are needed for :delete

sid-code committed:
Revert "Fix :delete command"

This reverts commit 98b17fbc34f13ad4a97a933d9120af12fb895fc4.

sid-code committed:
Revert "Erroneous lines shouldn't be added"

This reverts commit d24925160606d3a286c07c03721bf807021a1a39.

sid-code committed:
Merge branch 'master' of https://github.com/sid-code/nrpl

Fixing some mistakes

sid-code committed:
Revert "Revert "Erroneous lines shouldn't be added""

This reverts commit 0d54965eb0f799f06e17ff87d1539222e1d65434.

sid-code committed:
Revert "Modify compilation instructions"

This reverts commit 5086c909b334c1b4a023c4a644b3f46e666762c0.

sid-code commented:
I messed this up pretty bad. It should be all good now, though.

sid-code committed:
Remove inBlock variable, keep track of indent level instead

sid-code committed:
Add :indent command to manually change indent level

sid-code committed:
If var statement supplies a definition, don't indent

for example:

> var x = 6

shouldn't change the indent level

andreaferretti commented:
+1

This is the very first issue I encountered - first thing  I wrote was "1 + 3"

sid-code committed:
run with parallelbuild

sid-code committed:
Fix :delete command

sid-code committed:
Merge branch 'auto-indent'

sid-code committed:
Use echo instead of deprecated stdout.writeLn

sid-code committed:
Import sequtils for map

sid-code committed:
Improve actual REPL

Use rdstdin instead of stdin.readline
Actually indent lines that should be indented

sid-code committed:
Fix "echo" detection

Lines that START with echo shouldn't be added, not all lines that
contain echo

sid-code committed:
Increment version

sid-code committed:
Remove quit() at the end
